### PR TITLE
Added registration for VGS and replication CSI extension using updated csi-extensions

### PR DESCRIPTION
**Pull request by walker2**
# Description
Added registration for VGS and replication CSI extension using updated csi-extensions

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/128 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# Warning

This PR will be changed when https://github.com/dell/dell-csi-extensions/pull/13 is merged

### CODE REVIEW

**Review by maxan98 — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -26,11 +28,17 @@ require (
 	github.com/uber/jaeger-client-go v2.25.0+incompatible
 	github.com/uber/jaeger-lib v2.4.0+incompatible
 	golang.org/x/net v0.0.0-20210520170846-37e1c6afe023
-	google.golang.org/grpc v1.38.0
+	google.golang.org/grpc v1.42.0
```
Comment (harishp8889) [inline go.mod:31]: I saw the grpc version used in extensions is 1.38.0 https://github.com/dell/dell-csi-extensions/blob/e20060eb09e784a39b9eabe0e2646ea57f65e21f/replication/go.mod#L9 
Should we consider updating to 1.42.0 in extensions as well?

**Review by harishp8889 — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (walker2) [reply go.mod:31]: https://github.com/dell/dell-csi-extensions/blob/e20060eb09e784a39b9eabe0e2646ea57f65e21f/volumeGroupSnapshot/go.mod#L9

Comment (walker2) [reply go.mod:31]: It is 1.42 for VGS but 1.38 for replication, I can upgrade the replication one

Comment (walker2) [reply go.mod:31]: Done it in this commit https://github.com/dell/dell-csi-extensions/pull/13/commits/8ba8ce0b01c967b2cf3321fecdedc8a56fb9cb58

**Review by harishp8889 — APPROVED** [review]

**Comment by walker2** [comment]
@harishp8889 I think you merged it a little too early to the main, I haven't changed `replace` directive to new ones, I can't guarantee it will work. 
I specifically put a warning to do that

**Comment by walker2** [comment]
> @harishp8889 I think you merged it a little too early to the main, I haven't changed `replace` directive to new ones, I can't guarantee it will work. I specifically put a warning to do that

And we need also to create PRs to change both VGS and replication controller to use a newer version of extensions, otherwise sidecars will fail gRPC requests

**Pull request merged**